### PR TITLE
feat(metadata-client): add client for metadata service

### PR DESCRIPTION
**Pull request by jaredvu**
Add metadata service client

---

## Functions

* `lib/metadataService`
  * Add simple client for querying metadata

## Env var

* add `VITE_METADATA_SERVICE_URI`
---

### CODE REVIEW

File: src/lib/metadataService.ts [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+  _post(endpoint: string, params: {} = {}) {
+    return fetch(`${this.host}/${endpoint}`, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify(params),
+    })
+      .then((res) => res.json())
+      .then((data) => data);
```
Comment (tinaszheng) [inline src/lib/metadataService.ts:58]: does this line do anything

Comment (jaredvu) [reply src/lib/metadataService.ts:58]: I guess not since we aren't doing anything immediately with the data

File: src/lib/metadataService.ts [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+  }
+
+  async getMarketmap(): Promise<Record<string, string>> {
+    return this._post('marketmap-map');
+  }
+
+  async getAssetInfo({ assets }: { assets?: string[] }): Promise<MetadataServiceInfoResponse> {
```
Comment (tyleroooo) [inline src/lib/metadataService.ts:73]: Argument being an object isn't really doing much for us in this case.

**Review by tyleroooo — APPROVED** [review]

File: src/lib/metadataService.ts [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+  }
+
+  _post(endpoint: string, params: {} = {}) {
+    return fetch(`${this.host}/${endpoint}`, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify(params),
+    }).then((res) => res.json());
```
Comment (yogurtandjam) [inline src/lib/metadataService.ts:62]: [suggestion] add catch clauses

File: src/lib/metadataService.ts [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+  }
+
+  _get(endpoint: string) {
+    return fetch(`${this.host}/${endpoint}`).then((res) => res.json());
+  }
+
+  async getMarketmap(): Promise<Record<string, string>> {
+    return this._get('marketmap-map');
```
Comment (yogurtandjam) [inline src/lib/metadataService.ts:70]: [non blocking suggestion] convert string literals to str enums? i don't feel very opinionated about this one

**Review by tinaszheng — APPROVED** [review]

**Review by yogurtandjam — APPROVED** [review]
LGTM, some thoughts if u choose to implement them

**Comment by yogurtandjam** [comment]
general thought - should we be putting clients in a clients folder? or naming the files accordingly? 🤔

File: src/lib/metadataService.ts (hunk 3 above)
Comment (jaredvu) [reply src/lib/metadataService.ts:62]: I was going to utilize react-query `onError` to handle

Comment (jaredvu) [reply src/lib/metadataService.ts:62]: I'm also a fan of using `try/catch` wherever we actually utilize the fetches

Comment (yogurtandjam) [reply src/lib/metadataService.ts]: yeah that makes sense to me, i think if anything catches are best used within the client to throw more custom v4-web errors. but definitely not worried about that right now

**Pull request merged**